### PR TITLE
Golson pdx/5

### PR DESCRIPTION
**Pull request by golson-pdx**

### CODE REVIEW

File: tests/test_userhistory.py
Comment (amminer) [inline tests/test_userhistory.py:12]: I like this change, it's good to check on the None values and that revisions were pulled down and I feel that it makes sense here as opposed to getting its own test function - this is more a subroutine of __init__ than an independent behavior.

Regarding testing for the presence of revisions that should have been pulled down from the external API and included in user_history.revisions, I have been giving our test suite more thought lately and I have a lot of thoughts that tie in here. This would be an excellent time for me to learn how to use the github discussions feature. I'll try and do that after this review and maybe link back to this code to anchor a future conversation about the suite in general.

File: app.py
Comment (amminer) [inline app.py:37]: Consider using double quotes to keep consistent with the rest of the codebase

File: tests/test_userhistory.py
Comment (amminer) [reply tests/test_userhistory.py:12]: One of the main thoughts that I have is with regards to the default behavior we have right now, which pulls down every revision ever for a given user or article if no date range is specified. I think we should change this and will elaborate in a conversation, but for now I want to suggest that tests should include some level of start and end specificity to limit this testing overhead.

Additionally, when our default behavior changes, tests should specify consistent start and end datetimes so that we can guarantee that we get the expected responses back that allow the tests to pass, so I think this change is worth implementing either way. Unfortunately this does also muddy the filter tests a little bit since they would all then depend on the date/time code outside the filters they're supposed to be testing. We could potentially mock this in some way but this is starting to get highly complicated - I recall Daniel saying in a lecture that we should expect to spend as much as 2-3 hours writing tests for each hour we spend writing code, and I think this is a part of what he was talking about. If you have a bit of time today to implement this in the context of this PR that would be awesome, but I understand if it can't happen today and am happy to merge this as-is for the time being with the intent that we address this in all tests some time this last sprint. I'll get into it more in a conversation but I think we should spend significant time improving and building out our test suite, not only for the sake of the project overall but also as an exercise that's pretty relevant to the course material.

File: tests/test_userhistory.py
Comment (amminer) [inline tests/test_userhistory.py:25]: Given current network overhead and potential upcoming changes to default behavior, consider specifying consistent start and end dates/times

File: tests/test_userhistory.py
Comment (amminer) [inline tests/test_userhistory.py:16]: Given current network overhead and potential upcoming changes to default behavior, consider specifying consistent start and end dates/times.
Additionally it may be worth checking the expected behavior for a request that matches no revisions - this is something I tweaked recently in my code but failed to warn y'all about, sorry! Fortunately it didn't appear to break existing tests, but this is probably more of a condemnation of the thoroughness of our test suite than it is a happy footnote.

File: tests/test_userhistory.py
Comment (amminer) [inline tests/test_userhistory.py:1]: Consider using double quotes to keep consistent with the rest of the codebase -
I'm having some trouble here recalling whether we decided to use double or single quotes for docstrings. If we could all discuss this again briefly over discord or in a github conversation I would appreciate having it in writing to look back at in moments like these, sorry for being uncertain of where we landed in our verbal discussion.

File: tests/test_userhistory.py
Comment (amminer) [inline tests/test_userhistory.py:40]: Even if this type of date/time specification ends up in the other tests, I appreciate that it's covered for its own sake with this test.

**Review by amminer — COMMENTED** [review]
I tried to keep the comments as concise as possible but there were a couple of long ones, apologies in advance! I was hoping to get a couple of things addressed before approval, but if the test suite passes and the interface works then I am content to just merge for now and touch on these comments at a later date. The actual front end code is a little obscure to me but I'm looking forward to demoing it on my machine after this merges.

File: tests/test_userhistory.py
Comment (golson-pdx) [reply tests/test_userhistory.py:1]: Fixed this! All quotes will now be double

File: tests/test_userhistory.py
Comment (golson-pdx) [reply tests/test_userhistory.py]: Sounds good, leaving this test and keeping timestamps consistent with other tests

File: tests/test_userhistory.py
Comment (golson-pdx) [reply tests/test_userhistory.py]: Updated to use consistent timestamps and also check for no revision results

File: tests/test_userhistory.py
Comment (golson-pdx) [reply tests/test_userhistory.py]: Updated with consistent timestamps with other tests

File: tests/test_userhistory.py
Comment (golson-pdx) [reply tests/test_userhistory.py]: Updated to use timestamps that are consistent between all user history tests, we can further discuss implementing mocking in our tests

**Review by amminer — APPROVED** [review]
Looks good!

**Pull request merged**
